### DISCUSSION TITLE
Projectile Drop attributes

### DISCUSSION THREAD

**Discussion by JessicaTheGunLady** [nosoop/SM-TFCustAttr#3] [Created attributes]
_I'm doing this correctly, right?_

Two probably rather poorly coded attributes, one being `rocket launch`, causing a fired rocket to start to travel upwards, and `rocket drop`, making a fired rocket travel downwards.

Further details for the attributes as well as the plugin itself and the source code (if you dare to look at it) are available [here](https://github.com/JessicaTheGunLady/SM-TF2-ProjectileDrop).

**Comment by nosoop** [comment]
Yes, this is a fine place to put it.  Thank you for sharing!

Couple of constructive comments for the repo, if you're open to them:
- The workflow file operates on the `master` branch; the repository only has a `main` branch specified, so you'll want to change the workflow to specify `main` if you'd like Github Actions to run on that repo.
- The workflow doesn't do anything with the compiled code.  That's totally fine if you want to use it purely for catching compile-time errors, but Actions is capable of building releases as well.
- Git repositories don't handle binary files very well as every copy will be saved; I tend to use releases for that purpose, but other people do have their own opinions on that.  Keeping compiled plugins out of the repo also means you don't have to remember to it in lockstep with your source code changes.
- The idiomatic way to pass multiple things into a `RequestFrame` or other asynchronous callback is to use a `DataPack`. Might be tricky in this instance since you have to free it when the projectile is destroyed.
- As far as `TF2CustAttr_GetInt` being relatively expensive goes, this isn't too bad overall! It's way more of a concern for compounded things being run on every weapon on every client on every tick.
- `RequestFrame` is bound to tickrate. You may want to use [`GetGameFrameTime`](https://sm.alliedmods.net/new-api/halflife/GetGameFrameTime) to account for the true time taken.

**Reply by JessicaTheGunLady** [reply]
Always open to constructive comments, thanks.

The repository was set up basically unadjusted from how the Visual Studio Code Sourcemod plugin sets everything up and I kept it as it was because "hey, if it functions as it is, might as well just leave it", I'll make some adjustments as suggested though.

I'll also try using DataPacks, I had experimented with them before but was ultimately unable to get it working, I was probably doing something wrong though, I'll also try GetGameFrameTime.

Again, the comments are greatly appreciated, I'll be sure to mention here if I make anything else, as long as I don't end up making too many plugins and decide it's just worth merging it all into one repository.
